src/ivector/ivector-extractor.cc

```diff
@@ -1193,6 +1193,29 @@ double IvectorExtractorStats::Update(
   return ans;
 }
 
+void IvectorExtractorStats::IvectorVarianceDiagnostic(
+  const IvectorExtractor &extractor) {
+
+  // W is an estimate of the total residual variance explained by the
+  // speaker-adapated model.  B is an estimate of the total variance
+  // explained by the Ivector-subspace.
+  SpMatrix<double> W(extractor.Sigma_inv_[0].NumRows()),
+                      B(extractor.M_[0].NumRows());
+  Vector<double> w(gamma_);
+  w.Scale(1.0 / gamma_.Sum());
+  for (int32 i = 0; i < extractor.NumGauss(); i++) {
+    SpMatrix<double> Sigma_i(extractor.FeatDim());
+    extractor.InvertWithFlooring(extractor.Sigma_inv_[i], &Sigma_i);
+    W.AddSp(w(i), Sigma_i);
+    B.AddMat2(w(i), extractor.M_[i], kNoTrans, 1.0);
+  }
+  double trace_W = W.Trace(),
+         trace_B = B.Trace();
+  KALDI_LOG << "The proportion of variance explained by the iVectors "
+            << "versus the Gaussians is " << trace_B / (trace_B + trace_W)
```

Comment:
could you please reword to:
"The proportion of the within-Gaussian variance explained by the iVectors is "...